action-api/src/main/java/com/powsybl/action/GeneratorAction.java

```diff
@@ -82,4 +85,42 @@ public OptionalDouble getTargetV() {
     public OptionalDouble getTargetQ() {
         return targetQ == null ? OptionalDouble.empty() : OptionalDouble.of(targetQ);
     }
+
+    public NetworkModification toModification() {
+        // if values are outside of generator boundaries the boundaries are used instead
+        GeneratorModification.Modifs modifs = new GeneratorModification.Modifs();
+        modifs.setDoNotRestrictTargetPAndDoNotConnect(true);
```

Comment:
Then setting `doNotRestrictTargetPAndDoNotConnect` to `true`, targetP is modified without checking the value against the boundaries. The code is inconsistent with the comment here.

Reply:
fixed